File: LeetCode/Medium/0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> numberSt = new Stack<>();

        for (int i = 0; i < tokens.length; i++) {
            String token = tokens[i];
            int num1;
            int num2;
            switch (token) {
                case "+":
                    num1 = numberSt.pop();
                    num2 = numberSt.pop();
                    numberSt.push(num1 + num2);
                    break;
                case "-":
                    num1 = numberSt.pop();
                    num2 = numberSt.pop();
                    numberSt.push(num2 - num1);
                    break;
                case "*":
                    num1 = numberSt.pop();
                    num2 = numberSt.pop();
                    numberSt.push(num1 * num2);
                    break;
                case "/":
                    num1 = numberSt.pop();
                    num2 = numberSt.pop();
                    numberSt.push(num2 / num1);
                    break;
                default:
                    numberSt.push(Integer.parseInt(token));
                    break;
            }
        }
        return numberSt.pop();
    }
}

File: LeetCode/Medium/0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function (tokens) {
    const numbers = []

    for (let i = 0; i < tokens.length; i++) {
        let num1, num2;
        let token = tokens[i]

        switch (token) {
            case "+":
                num1 = numbers.pop()
                num2 = numbers.pop()
                numbers.push(num2 + num1)
                break;
            case "-":
                num1 = numbers.pop()
                num2 = numbers.pop()
                numbers.push(num2 - num1)
                break;
            case "/":
                num1 = numbers.pop()
                num2 = numbers.pop()
                numbers.push(Math.trunc(num2 / num1))
                break;
            case "*":
                num1 = numbers.pop()
                num2 = numbers.pop()
                numbers.push(num2 * num1)
                break;
            default:
                numbers.push(parseInt(token))
                break;
        }
    }
    return numbers.pop()
};